app/main.py

```diff
@@ -1,2 +1,91 @@
+from typing import Optional, Any, Hashable
+
+
+class Node:
+    def __init__(
+            self,
+            key: Any,
```

Comment:
Indicate that the `key` should be `Hashable` in all places in the type annotation where you expect to obtain it, not only in the `__setitem__`.